packages/logger/src/Logger.ts

```diff
@@ -1142,30 +1180,24 @@ class Logger extends Utility implements LoggerInterface {
    * @param sampleRateValue - The sample rate value
    */
   private setInitialSampleRate(sampleRateValue?: number): void {
-    this.powertoolsLogData.sampleRateValue = 0;
     const constructorValue = sampleRateValue;
     const customConfigValue =
       this.getCustomConfigService()?.getSampleRateValue();
     const envVarsValue = this.getEnvVarsService().getSampleRateValue();
     for (const value of [constructorValue, customConfigValue, envVarsValue]) {
       if (this.isValidSampleRate(value)) {
+        this.#debugLogSampling.sampleRateValue = value;
         this.powertoolsLogData.sampleRateValue = value;
 
         if (
-          this.logLevel > LogLevelThreshold.DEBUG &&
-          value &&
-          randomInt(0, 100) / 100 <= value
+          this.#shouldEnableDebugSampling() &&
+          this.logLevel > LogLevelThreshold.TRACE
         ) {
-          // only change logLevel if higher than debug, i.e. don't change from e.g. tracing to debug
           this.setLogLevel('DEBUG');
           this.debug('Setting log level to DEBUG due to sampling rate');
-        } else {
-          this.setLogLevel(
-            this.getLogLevelNameFromNumber(this.#initialLogLevel)
-          );
         }
 
-        return;
+        break;
       }
```

Comment:
Just curious why break instead of return, it has the same effect here

Reply:
Intent is clearer with a break, if we ever add some more code below it'll run as we expected, rather than being skipped as it would with a return.